redash/query_runner/db2.py

```diff
@@ -0,0 +1,147 @@
+import os
+import json
+import logging
+import select
+
+import ibm_db_dbi
+
+from redash.query_runner import *
+from redash.utils import JSONEncoder
+
+logger = logging.getLogger(__name__)
+
+types_map = {
+    ibm_db_dbi.NUMBER: TYPE_INTEGER,
+    ibm_db_dbi.BIGINT: TYPE_INTEGER,
+    ibm_db_dbi.ROWID: TYPE_INTEGER,
+    ibm_db_dbi.FLOAT: TYPE_FLOAT,
+    ibm_db_dbi.DECIMAL: TYPE_FLOAT,
+    ibm_db_dbi.DATE: TYPE_DATE,
+    ibm_db_dbi.TIME: TYPE_DATETIME,
+    ibm_db_dbi.DATETIME: TYPE_DATETIME,
+    ibm_db_dbi.BINARY: TYPE_STRING,
+    ibm_db_dbi.XML: TYPE_STRING,
+    ibm_db_dbi.TEXT: TYPE_STRING,
+    ibm_db_dbi.STRING: TYPE_STRING
+}
+
+
+class DB2(BaseSQLQueryRunner):
+    noop_query = "SELECT 1 FROM SYSIBM.SYSDUMMY1"
+
+    @classmethod
+    def configuration_schema(cls):
+        return {
+            "type": "object",
+            "properties": {
+                "user": {
+                    "type": "string"
+                },
+                "password": {
+                    "type": "string"
+                },
+                "host": {
+                    "type": "string",
+                    "default": "127.0.0.1"
+                },
+                "port": {
+                    "type": "number",
+                    "default": 50000
+                },
+                "dbname": {
+                    "type": "string",
+                    "title": "Database Name"
+                }
+            },
+            "order": ['host', 'port', 'user', 'password', 'dbname'],
+            "required": ["dbname"],
+            "secret": ["password"]
+        }
+
+    @classmethod
+    def type(cls):
+        return "db2"
+
+    @classmethod
+    def enabled(cls):
+        try:
+            import ibm_db
+        except ImportError:
+            return False
+
+        return True
+
+    def _get_definitions(self, schema, query):
+        results, error = self.run_query(query, None)
+
+        if error is not None:
+            raise Exception("Failed getting schema.")
+
+        results = json.loads(results)
+
+        for row in results['rows']:
+            if row['TABLE_SCHEMA'] != 'public':
+                table_name = '{}.{}'.format(row['TABLE_SCHEMA'], row['TABLE_NAME'])
```

Comment:
Please create a unicode string here (i.e. prefix with `u'` instead of `'`).

Reply:
OK, added